src/Analyzers/CSharp/Analyzers/UseParameterNullChecking/CSharpUseParameterNullCheckingDiagnosticAnalyzer.cs

```diff
@@ -141,15 +141,15 @@ protected override void InitializeWorker(AnalysisContext context)
 
             foreach (var statement in block.Statements)
             {
-                var parameter = TryGetParameterNullCheckedByStatement(statement);
+                var parameter = TryGetParameterNullCheckedByStatement(statement, out var location);
                 if (ParameterCanUseNullChecking(parameter)
                     && parameter.DeclaringSyntaxReferences.FirstOrDefault() is SyntaxReference reference
                     && reference.SyntaxTree.Equals(statement.SyntaxTree)
                     && reference.GetSyntax() is ParameterSyntax parameterSyntax)
                 {
                     context.ReportDiagnostic(DiagnosticHelper.Create(
                         Descriptor,
-                        statement.GetLocation(),
+                        location ?? statement.GetLocation(),
```

Comment:
```suggestion
                        diagnosticLocation ?? statement.GetLocation(),
```

Reply:
honestly i don't love this.  just have TryGetParameterNullCheckedByStatement always set the diagnosticLocation.